src/hdmf/container.py

```diff
@@ -407,13 +408,32 @@ def set_dataio(self, **kwargs):
         dataio.data = self.__data
         self.__data = dataio
 
+    @docval({'name': 'func', 'type': types.FunctionType, 'doc': 'a function to transform *data*'})
+    def transform(self, **kwargs):
+        """
```

Comment:
Any transformation to the underlying data may change subsequent behavior. Users should be warned about this in the documentation string